### PR TITLE
fix: do not restart standard exit that waits in exit processing queue

### PR DESCRIPTION
**Pull request by pgebal**
Closes https://github.com/omisego/security-issues/issues/8
* Prevents user from restarting an exit that is enqueued in exit processing queue.
* Adds tests for missing branches in `PaymentInFlightExitRouter.sol` and `PaymentStandardExitRouter.sol`

### CODE REVIEW

File: plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js [hunk 1]
```diff
@@ -144,6 +149,11 @@ contract('PaymentStandardExitRouter', ([_, alice]) => {
                 'ExitOmitted',
                 { exitId: new BN(exitId) },
             );
+
+            const exitData = (await this.exitGame.standardExits([exitId]))[0];
+            Object.values(exitData).forEach((val) => {
```
Comment (kevsul) [inline plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js]: Same as above

File: plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js [hunk 2]
```diff
@@ -129,6 +129,11 @@ contract('PaymentStandardExitRouter', ([_, alice]) => {
...
```
[the remark this answers is not in the document]
Comment (boolafish) [reply plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js]: It is not the same as what you propose though. It is checking each field is any of the defined "empty" result. It is a looser test than what you want to propose though. However, I remember the returned result would be a weird js array with some extra key value pairs.....

File: plasma_framework/contracts/src/exits/payment/controllers/PaymentStartStandardExit.sol [hunk 3]
```diff
@@ -177,7 +177,8 @@ library PaymentStartStandardExit {
         require(data.outputGuardHandler.getExitTarget(data.outputGuardData) == msg.sender, "Only exit target can start an exit");
 
         require(data.controller.txFinalizationVerifier.isStandardFinalized(data.finalizationData), "The transaction must be standard finalized");
-        require(exitMap.exits[data.exitId].exitable == false, "Exit has already started");
+        PaymentExitDataModel.StandardExit memory exit = exitMap.exits[data.exitId];
+        require(!exit.exitable && exit.amount == 0, "Exit has already started");
```
[the remark this answers is not in the document]
Comment (pgebal) [reply plasma_framework/contracts/src/exits/payment/controllers/PaymentStartStandardExit.sol]: You are right.

File: plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevsul) [reply plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js]: Yeah, I was thinking that the current way was too loose, so e.g. we're expecting `outputId` to be EMPTY_BYTES_32, but if it was `0` or `false` somehow then the test would still pass.
But I guess that should be tested elsewhere anyway

[the remark this answers is not in the document]
Comment (pgebal) [reply plasma_framework/test/src/exits/payment/PaymentProcessStandardExit.test.js]: works great!

**Pull request merged**